soem/ethercatmain.c

```diff
@@ -724,11 +724,19 @@ int ecx_readstate(ecx_contextt *context)
    ec_alstatust sl[MAX_FPRD_MULTI];
    uint16 slca[MAX_FPRD_MULTI];
    boolean noerrorflag, allslavessamestate;
+   boolean allslavespresent = FALSE;
+   int wkc;
 
    /* Try to establish the state of all slaves sending only one broadcast datargam.
     * This way a number of datagrams equal to the number of slaves will be sent only if needed.*/
    rval = 0;
-   ecx_BRD(context->port, 0, ECT_REG_ALSTAT, sizeof(rval), &rval, EC_TIMEOUTRET);
+   wkc = ecx_BRD(context->port, 0, ECT_REG_ALSTAT, sizeof(rval), &rval, EC_TIMEOUTRET);
+
+   if(wkc == *(context->slavecount))
```

Comment:
Should make == into >=. There can be the case that an extra slave is added to the network. So wkc can be larger than slavecount

Reply:
Will do